CMakeLists.txt

```diff
@@ -1,332 +1,387 @@
     endif()
+    
+    add_library(bmi_fortran_compat SHARED IMPORTED)
+    set_target_properties(bmi_fortran_compat PROPERTIES IMPORTED_LOCATION "${BMI_FORTRAN_ISO_C_LIB_PATH}")
 endif()
 
-# If Python set to be active, run steps to make sure it is included properly, and set compiler definition to true
-if (${NGEN_ACTIVATE_PYTHON})
+# -----------------------------------------------------------------------------
+if(NGEN_WITH_PYTHON)
     add_compile_definitions(ACTIVATE_PYTHON=true)
-
     find_package(Python 3.6.8 REQUIRED COMPONENTS Interpreter Development NumPy)
-    #message(INFO " the final Py lib is ${Python_LIBRARY}")
-    #message(INFO " the final Py libraries are ${Python_LIBRARIES}")
-    #message(INFO " the final Py library dirs are ${Python_LIBRARY_DIRS}")
-    #message(INFO " the final Py runtime library dirs are ${Python_RUNTIME_LIBRARY_DIRS}")
-    #message(INFO " the final Py std lib is ${Python_STDLIB}")
-    #message(INFO " the rpath is ${CMAKE_INSTALL_RPATH}")
     set(PYTHON_EXECUTABLE ${Python_EXECUTABLE}) # Case-sensitive difference
     add_subdirectory(extern/pybind11)
 endif()
 
-target_include_directories(ngen PUBLIC
-        "${PROJECT_BINARY_DIR}/include" # For generated config header file in binary tree
-        )
-
-  
-# In some IDEs (e.g., CLion), it is easier to work with the code in the included example of an "external" shared library
-# when that code appears to be part of the main project, which can be done by add a statement like the ones below.
-#
-# However, these should be removed or commented out when running CMake builds and in the committed version of this file.
-#
-#add_subdirectory("extern/cfe")
-#add_subdirectory("extern/test_bmi_c")
-
-if(QUIET)
-set(UDUNITS_QUIET true)
-set(ET_QUIET true)
-set(NGEN_QUIET true)
+# -----------------------------------------------------------------------------
+if(NGEN_WITH_ROUTING)
+    add_compile_definitions(NGEN_ROUTING_ACTIVE)
+endif()
+
+# -----------------------------------------------------------------------------
+if(NGEN_QUIET)
+    set(UDUNITS_QUIET true)
+    set(ET_QUIET true)
+    add_compile_definitions(NGEN_QUIET)
 endif()
 
 if(UDUNITS_QUIET)
-add_compile_definitions(UDUNITS_QUIET)
+    add_compile_definitions(UDUNITS_QUIET)
 endif()
 
+# FIXME: I don't think this is needed now? - Justin
 if(ET_QUIET)
-add_compile_definitions(ET_QUIET)
+    add_compile_definitions(ET_QUIET)
 endif()
 
-if(NGEN_QUIET)
-add_compile_definitions(NGEN_QUIET)
-endif()
+# -----------------------------------------------------------------------------
 
-add_subdirectory("src/core")
-add_dependencies(core libudunits2)
-add_subdirectory("src/geojson")
+# Natively support BMI C++ modules and pre-compile in the test_bmi_cpp mock/example.
+set(TEST_BMI_CPP_DIR ${NGEN_EXT_DIR}/test_bmi_cpp)
+git_update_submodule(${NGEN_EXT_DIR}/bmi-cxx)
+add_subdirectory(${TEST_BMI_CPP_DIR} ${TEST_BMI_CPP_DIR}/cmake_build) # FIXME: Why is this here? - Justin
```

Comment:
Does the test get built if this isn't here? I would think not

Reply:
Right, I think I put this comment because I intended to move into the `test/CMakeLists.txt`, since it doesn't make sense to build it unless tests are being built -- if that makes sense

Reply:
If tests are enabled, it may make sense to have flags (that are dependent on tests being enabled) for building the various test BMI models in this same fashion. i.e. if we want to build the fortran test bmi model when configuring for tests, then we could pass `-DNGEN_BUILD_FORTRAN_TEST_BMI` or something similar

Reply:
If it's easy to square away, go ahead and do it in this PR. If it would take more than a few minutes of additional effort to figure out, let's just leave it be for now.

Reply:
Good call, I think for the sake of this PR, I'll move this code block into the test CMakeLists as is, and then handle those extra options in a future PR.